Responsive Dashboard

amitojsingh opened the pull request:
1. Responsive Dashboard
2. Responsive Word Pages
3. Responsive Question Pages

MySmilingTurban commented:
The words you learnt box collapses when width goes below 768

https://github.com/KhalisFoundation/gurmukhi-web/assets/51584037/87a95a49-dc8e-417f-af8e-b4b04b0f68da

MySmilingTurban commented:
Another thing, below 341 width, it squeezes the profile icon in the header
Also, make sure the background remains as parallax because right now I can see it repeat

https://github.com/KhalisFoundation/gurmukhi-web/assets/51584037/bf301324-a976-450a-bf2c-6acd600197e3

amitojsingh commented:
Fixed @MySmilingTurban  Veer ji can you try now.

amitojsingh commented:
@MySmilingTurban  Veer ji I changed the design because the text was getting overlapped with the header on scroll which is poor typography so to avoid that I have added a background to header and footer which is very much similar with minor difference.

MySmilingTurban commented:
Please don't add background image to header and footer, it messes up the design

https://github.com/KhalisFoundation/gurmukhi-web/assets/51584037/a59a8774-1b9d-40b9-bb8e-23e4c41d870f

MySmilingTurban commented:
Oh, can you can there background to blur instead of a solid picture, cuz it feels odd

MySmilingTurban commented:
you can remove background picture from footer, as it will not overlap any text as it is fixed to bottom of the page

amitojsingh commented:
fixed veer ji

MySmilingTurban commented on src/components/questions/multiple-choice.tsx:
```diff
@@ -107,7 +107,7 @@ export default function MultipleChoiceQuestion({
     <div className='flex flex-col items-left justify-evenly text-center'>
       <div className='flex flex-row items-center justify-between gap-5 rounded-lg p-4'>
-        <h1 className={'text-5xl gurmukhi text-black font-semibold'}>
+        <h1 className={'xl:text-5xl text-4xl gurmukhi text-black font-semibold leading-snug'}>
```
nit: media query class name should be written after general ones

amitojsingh replied:
fixed

MySmilingTurban commented on src/assets/icons/Shabadavali.tsx:
```diff
@@ -4,8 +4,8 @@ import { useTranslation } from 'react-i18next';
   return (
-    <div className="flex items-center justify-center w-12 h-12 mr-4 rounded-full logo-bg">
-      <span className="text-2xl text-white gurmukhi mustard">{text('SHASHA')}</span>
+    <div className='flex items-center justify-center md:w-12 md:h-12 w-5 h-5 mr-4 rounded-full logo-bg'>
```
media query class name should be written after general ones

amitojsingh replied:
fixed

MySmilingTurban commented on src/pages/dashboard/components/coinbox/index.tsx:
```diff
@@ -11,10 +11,10 @@ function CoinBox({ commonStyle }: { commonStyle: string }) {
     <div className={commonStyle}>
       <div className='flex justify-center items-center h-full'>
-        <div className='my-4'>
+        <div className=''>
```
nit: remove empty classname

amitojsingh replied:
fixed

MySmilingTurban commented on src/pages/wincoin/index.tsx:
```diff
@@ -59,7 +59,7 @@ function WinCoin() {
       <Meta title={title} description={description} />
       <div className='flex flex-col text-center justify-evenly h-4/5 recoleta'>
-        <div className=' mx-auto bg-cyan-50 w-1/4 rounded-xl'>
+        <div className=' mx-auto bg-cyan-50 xl:w-1/4 w-2/4 rounded-xl'>
```
nit: media query class name should be written after general ones

amitojsingh replied:
fixed

MySmilingTurban commented on src/pages/word/information/index.tsx:
```diff
@@ -78,16 +78,16 @@ export default function Information() {
     <div className='flex flex-col items-center w-full h-full justify-between gap-5'>
       <Meta title={title} description={description} />
-      <div className='flex flex-col h-3/4 justify-between items-center'>
+      <div className='flex flex-col h-3/4 justify-between items-center '>
```
nit: empty space

amitojsingh replied:
fixed

MySmilingTurban commented on src/pages/word/information/index.tsx:
```diff
@@ -78,16 +78,16 @@ export default function Information() {
-        <div className='flex flex-row items-center justify-between h-full gap-10 w-full'>
-          <div className='flex flex-col items-left justify-evenly w-1/2 h-full'>
+        <div className='flex flex-col xl:flex-row items-center justify-between h-full gap-10 w-full py-10 overflow-y-auto'>
+          <div className='flex flex-col items-left justify-evenly xl:w-1/2 w-3/4 h-full'>
```
nit: media query class name should be written after general ones

amitojsingh replied:
fixed

MySmilingTurban commented on src/pages/word/information/index.tsx:
```diff
@@ -146,7 +148,7 @@ export default function Information() {
             <div className='flex items-center justify-around gap-5 w-full h-1/2'>
               <div
-                className={`w-2/5 h-full cardImage bg-cover bg-sky-100 bg-blend-soft-light hover:bg-sky-50 border-2 border-sky-200 shadow-lg rounded-lg ${
+                className={`lg:w-2/5 w-5/6 h-full cardImage bg-cover bg-sky-100 bg-blend-soft-light hover:bg-sky-50 border-2 border-sky-200 shadow-lg rounded-lg ${
```
nit: media query class name should be written after general ones

amitojsingh replied:
fixed

MySmilingTurban commented on src/pages/word/information/index.tsx:
```diff
@@ -172,7 +174,7 @@ export default function Information() {
               </div>
               <div
-                className={`w-2/5 h-full cardImage bg-cover bg-sky-100 bg-blend-soft-light hover:bg-sky-50 border-2 border-sky-200 shadow-lg rounded-lg ${
+                className={`lg:w-2/5 w-5/6 h-full cardImage bg-cover bg-sky-100 bg-blend-soft-light hover:bg-sky-50 border-2 border-sky-200 shadow-lg rounded-lg ${
```
nit: media query class name should be written after general ones

amitojsingh replied:
fixed

MySmilingTurban commented on src/styles/globals.scss:
```diff
@@ -30,18 +32,23 @@ html,body,#root{
 * {
   font-family: 'Recoleta';
 }
+.header-bg{
+  background:rgb(244 250 255 / 81%);
+}
+.back-bg {
+  background: rgb(184 214 234);
+}
+.back-bg-image {
```
nit: add empty line after each class style

amitojsingh replied:
fixed

MySmilingTurban commented on src/assets/icons/Shabadavali.tsx:
```diff
@@ -4,8 +4,8 @@ import { useTranslation } from 'react-i18next';
   return (
-    <div className="flex items-center justify-center w-12 h-12 mr-4 rounded-full logo-bg">
-      <span className="text-2xl text-white gurmukhi mustard">{text('SHASHA')}</span>
+    <div className='flex items-center justify-center w-5 h-5 mr-4 rounded-full logo-bg md:w-12 md:h-12 '>
```
nit: remove empty space in end of classname

amitojsingh replied:
fixed

MySmilingTurban commented on src/components/questions/multiple-choice.tsx:
```diff
@@ -107,7 +107,7 @@ export default function MultipleChoiceQuestion({
     <div className='flex flex-col items-left justify-evenly text-center'>
       <div className='flex flex-row items-center justify-between gap-5 rounded-lg p-4'>
-        <h1 className={'text-5xl gurmukhi text-black font-semibold'}>
+        <h1 className={' gurmukhi text-black font-semibold leading-snug xl:text-5xl text-4xl'}>
```
change the classname to 
`gurmukhi text-black font-semibold leading-snug text-4xl xl:text-5xl`
(remove empty space and order of media query)

amitojsingh replied:
fixed

MySmilingTurban commented:
Please fix the image pages for word

<img width="769" alt="Screenshot 2024-03-02 at 5 51 55 pm" src="https://github.com/KhalisFoundation/gurmukhi-web/assets/51584037/93e173d6-aec0-46fd-b632-4449354a942a">

amitojsingh commented:
@MySmilingTurban veer ji looks like you have zoomed in on the image can you verify with 100% zoom?

amitojsingh commented:
@MySmilingTurban Veer ji please review

MySmilingTurban reviewed (changes requested):
Looks good to me!
One minor issue with the highlighted element in the attached screenshot, can we make it responsive too

<img width="438" alt="Screenshot 2024-03-10 at 10 20 23 am" src="https://github.com/KhalisFoundation/gurmukhi-web/assets/51584037/ca028bf5-f669-477a-b0e8-1ccd9bcc4a89">

amitojsingh commented:
fixed veer ji @MySmilingTurban

MySmilingTurban reviewed (approved):
Looks good to me!
